### DISCUSSION TITLE
Question: ALE-Kondo-Integration: Was Kondo always non-destructive/invasive against the source files

### DISCUSSION THREAD

**Discussion by christoph-frick** [clj-kondo/clj-kondo#1668] [Q&A] [answered]
[ALE](https://github.com/dense-analysis/ale) lints code inside vim.  There is integration for kondo already.  Kondo v2022.04.08 re-introduced `:namespace-name-mismatch` which was [removed](https://github.com/clj-kondo/clj-kondo/commit/af6b18c03ab9603a96cd33fe0b8bc33d8c934675) due to problems with windows.

Back when it was introduced first it led to problems with ALE, which I addressed with a [PR against ALE](https://github.com/dense-analysis/ale/pull/4109) but then redrew due to the original problem being gone (for now) in Kondo and concerns of the author.

The current ALE-Kondo-integration uses a *copy* of the current source file (under `/tmp` with random naming, different hierarchy).  This is basically the default for ALE so linters/... can do *destructive* things on the source.

My PR changed this to just use the *actual* source file (no copy).  The ALE authors concern back then was, that changing this might break integration with old Kondo versions, that relied on the copy.

To my knowledge Kondo always has been used to just read source ideally inside the project for the most insight and then only report it's findings.  It was never destructive/invasive.  Is this correct?

Thanks for any hints.

**Comment by borkdude** [comment]
Perhaps @lispyclouds knows something about this?

**Comment by christoph-frick** [comment]
I think I know now why the "temp approach" was chosen and is "better". This also works while the file is edited or a new file is not yet written. With my proposed ALE change this would only allow the linting to take place after saving the file, which would be less responsive for the developer.

The question itself remains, but I think this has to be solved with a different approach with ALE.

**Comment by lispyclouds** [comment, marked as answer]
@christoph-frick yes I don't think clj-kondo does any modifications to the file(s) its linting and it never did; its only concerned about reporting the findings (@borkdude can maybe confirm?) . Modifications via clj-kondo _may_ be done by tooling that embeds it like [clojure-lsp](https://clojure-lsp.io/).

**Reply by borkdude** [reply]
Correct, clj-kondo never makes any modifications.

**Reply by christoph-frick** [reply]
Thanks.  This was the answer I was looking for.
